plugins/sh/src/com/intellij/sh/run/ShRunConfiguration.java

```diff
@@ -64,6 +65,10 @@ public void checkConfiguration() throws RuntimeConfigurationException {
   @Nullable
   @Override
   public RunProfileState getState(@NotNull Executor executor, @NotNull ExecutionEnvironment environment) {
+    RunProfileState profileState = ShSupport.getInstance().createRunProfileState(executor, environment);
+    if (profileState != null) {
+      return profileState;
+    }
     return new ShRunConfigurationProfileState(getProject(), this);
```

Comment:
Maybe it's better to move `ShRunConfigurationProfileState` initialization into `ShSupport.getInstance().createRunProfileState`? and mark it as `@NotNull`?